app/routes.py

```diff
@@ -1,2 +1,184 @@
+from app import db
+from app.models.task import Task
+from sqlalchemy import asc, desc
+from datetime import datetime
+import requests
+import os
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["GET","POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort = request.args.get("sort")
+
+        if sort == "asc":
+            tasks = Task.query.order_by(asc("title"))
+        
+        elif sort == "desc":
+            tasks = Task.query.order_by(desc("title"))
+        
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.is_complete()
+            })
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        if "title" not in request_body.keys() or "description" not in request_body.keys() or "completed_at" not in request_body.keys():
+            return make_response({"details": "Invalid data"}, 400)
+
+        else:    
+            new_task = Task(title=request_body["title"], description=request_body["description"], completed_at=request_body["completed_at"])
+            
+            db.session.add(new_task)
+            db.session.commit()
+
+            return make_response({"task": new_task.make_json()}, 201)
+
+@tasks_bp.route("/<task_id>", methods= ["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+
+    elif request.method == "GET":
+        return make_response({"task": task.make_json()})
+
+    elif request.method == "PUT":
+        request_body = request.get_json()
+
+        task.title = request_body["title"]
+        task.description = request_body["description"]
+        task.completed_at = request_body["completed_at"]
+
+        db.session.commit()
+
+        return make_response({"task": task.make_json()}) 
+        
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+
+        return {
+            "details": (f"Task {task.task_id} \"{task.title}\" successfully deleted")
+        }
+
+@tasks_bp.route("/<task_id>/mark_complete", methods= ["PATCH"])
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return make_response("", 404)
+
+    API_KEY = os.environ.get("API_KEY")
+    PATH = "https://slack.com/api/chat.postMessage"
+    query_params = {
+            "channel": "task-notifications",
+            "text": f"Someone just completed the task {task.title}."
+        }
+    task.completed_at = datetime.utcnow()
+    db.session.commit()
+    requests.post(PATH, data=query_params, headers={"Authorization":API_KEY})
```

Comment:
Consider moving this functionality into a helper function increase readability and changeability.